examples/c/mnist_slp.c

```diff
@@ -1,67 +1,41 @@
-#include <stdio.h>
+#include <assert.h>
 
 #include <crystalnet.h>
 
-// y = softmax(xw + b)
-model_t *slp_model(shape_t *image_shape, int arity)
+// y = softmax(flatten(x) * w + b)
+s_model_t *slp(const shape_t *image_shape, uint8_t arity)
 {
-    shape_t *lable_shape = make_shape(1, arity);
-    shape_t *weight_shape = make_shape(2, shape_dim(image_shape), arity);
-    shape_t *x_wrap_shape = make_shape(1, shape_dim(image_shape));
+    shape_ctx_t *sc = new_shape_ctx();
+    s_model_ctx_t *ctx = make_s_model_ctx();
 
-    model_ctx_t *m = new_model_ctx();
-    node_t *x_ = make_placeholder(m, image_shape);
-    node_t *x = wrap_node(m, x_wrap_shape, x_);
-    node_t *w = make_parameter(m, weight_shape);
-    node_t *b = make_parameter(m, lable_shape);
+    symbol x = var(ctx, image_shape);
+    symbol x_ = reshape(ctx, mk_shape(sc, 1, shape_dim(image_shape)), x);
+    symbol w = covar(ctx, mk_shape(sc, 2, shape_dim(image_shape), arity));
+    symbol b = covar(ctx, mk_shape(sc, 1, arity));
```

Comment:
@zsdonghao Is it easy for you to understand co-variable and variable?

In crystalnet, variable is equivalent to the placeholders in TF; while co-variable is equivalent to variables in TF.

Reply:
In the future, complex models will be constructed using layer based APIs, there will be no `covar` any more.